R/01-qc.R

```diff
@@ -1,39 +1,119 @@
 
-#' This is a title for a function
+#' A function to run QC
 #' @description This is a function here's where we describe what it does
-#' @param parameter Here's a parameter let's describe it here
+#' @param plasmid_cutoff default it NULL; `qc_plasmid_histogram` will calculate and set a default one unless a value is specified here 
+#' @param use_combined default it TRUE; if TRUE, both zero count and low plasmid CPM filters are applied and if either is TRUE, a pgRNA construct will be filtered out. If FALSE, need to allow to specify which should be used
+#' @param params default is `list(cell_line = NULL)`; used to pass info when saving plots
+#' @param plots_dir default is `./qc_plots`; directory to save plots created with this function, if it doesn't exist already it will be created
+#' @param overwrite default is FALSE; whether to overwrite the QC Report file
+#' @param output_format default is NULL; the output format that the QC Rmd will be knitted to/produced
+#' @param output_file default is `QC_Report`; name of the output QC report file
+#' @param output_dir default is `./`; where the qc report will be stored
+#' @param wide_ar default is 0.75; aspect ratio
 #' @export
 #' @importFrom tidyr pivot_longer
-#' @import ggplot2
+#' @importFrom magrittr %>%
 #' @examples \dontrun{
 #'
 #' }
 
-run_qc <- function(gimap_data, plots_dir = "./qc_plots", wide_ar = 0.75, square_ar = 1) {
 
-  if (!dir.exists(plots_dir)) {
+run_qc <- function(gimap_dataset, plasmid_cutoff = NULL, use_combined = TRUE,
+                   params = list(cell_line = NULL), plots_dir = "./qc_plots", overwrite = FALSE, output_format = NULL, output_file = "QC_Report", output_dir = "./",
+                   wide_ar = 0.75) {
+  
+  store_results <- function(result_obj, lvoi, list_of_qc_things, plots_dir, params){
```

Comment:
This function should be broken out into its own so it can have documentation about it. For example, I'm not sure what `lvoi` is representing.